src/fdonotification.c

```diff
@@ -110,13 +113,24 @@ notify_signal (GDBusConnection *connection,
     {
       g_variant_get (parameters, "(u&s)", &id, &action);
     }
+  else if (g_str_equal (signal_name, "ActivationToken") &&
+           g_variant_is_of_type (parameters, G_VARIANT_TYPE ("(us)")))
+    {
+      g_variant_get (parameters, "(u&s)", &id, &activation_token);
+    }
   else
     return;
 
   n = fdo_find_notification_by_notify_id (id);
   if (n == NULL)
     return;
 
+  if (activation_token)
+    {
+      n->activation_token = g_strdup (activation_token);
```

Comment:
Is there a chance that `n->activation_token` may have already been set on a notification? Because if that's the case, you may want to do:

```suggestion
      g_clear_pointer (&n->activation_token, g_free);
      n->activation_token = g_strdup (activation_token);
```

instead.

Reply:
Shouldn't be set every, but who knows if something calls the `ActivationToken` multiple times on the same notification.